### PR TITLE
Add variant: add support for generic F446Z(C-E)(H-J-T) and Nucleo-F446ZE

### LINKED ISSUE #1487: Readme.md should be updated to show support for variant F446ZE
The readme.md file ([here](https://github.com/stm32duino/Arduino_Core_STM32/blob/master/README.md)) does not list the STM32F446ZE as a supported board. However, 5 months ago this folder was added: 

https://github.com/stm32duino/Arduino_Core_STM32/tree/2.0.0/variants/STM32F4xx/F446Z(C-E)(H-J-T)

I believe this means the F446ZE is supported, but the readme.md file hasn't been updated to show this.

### PR DESCRIPTION
**Pull request by CMGeldenhuys**
**Summary**

This PR adds support for the STM32F446Z(C-E)(H-J-T) variants as well as the Nucleo-F446ZE board. The code has been tested on a Nucleo-F446ZE.

**Closing issues**

Closes #1487 (Already marked as closed, but no support was added, from what I can see.)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,6 +91,7 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 | :green_heart:  | STM32F412ZG | [Nucleo F412ZG](http://www.st.com/en/evaluation-tools/nucleo-f412zg.html) | *2.6.0* |  |
 | :green_heart:  | STM32F413ZH | [Nucleo F413ZH](http://www.st.com/en/evaluation-tools/nucleo-f413zh.html) | *2.4.0* |  |
 | :green_heart:  | STM32F429ZI | [Nucleo F429ZI](http://www.st.com/en/evaluation-tools/nucleo-f429zi.html) | *0.1.0* |  |
+| :yellow_heart:  | STM32F446ZE | [Nucleo F446ZE](http://www.st.com/en/evaluation-tools/nucleo-f446ze.html) | *2.7.0* |  |
```
Comment (fpistm) [inline README.md:94]:
```suggestion
| :yellow_heart:  | STM32F446ZE | [Nucleo F446ZE](http://www.st.com/en/evaluation-tools/nucleo-f446ze.html) | **2.7.0** |  |
```

File: README.md [hunk 2]
```diff
@@ -368,6 +369,7 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 | :green_heart: | STM32F437ZG<br>STM32F437ZI | Generic Board | **2.0.0** |
 | :green_heart: | STM32F439ZG<br>STM32F439ZI | Generic Board | **2.0.0** |
 | :green_heart:  | STM32F446RC<br>STM32F446RE | Generic Board | *1.9.0* |  |
+| :yellow_heart:  | STM32F446ZC<br>STM32F446ZE | Generic Board | *2.7.0* |  |
 | :green_heart: | STM32F446VC<br>STM32F446VE | Generic Board | **2.0.0** |
```
Comment (fpistm) [inline README.md:373]:
```suggestion
| :green_heart: | STM32F446VC<br>STM32F446VE | Generic Board | *2.0.0* |
| :yellow_heart:  | STM32F446ZC<br>STM32F446ZE | Generic Board | **2.7.0** |  |
```

**Review by fpistm — CHANGES_REQUESTED** [review]
Hi @CMGeldenhuys 
Thanks for this PR.
Sorry  for the delay, I was on vacation.
First quick review. Let's CI run to see if any errors occurs.

**Comment by CMGeldenhuys** [comment]
Hi @fpistm, I've updated the README as requested and squashed the changes into the old commits.

**Pull request merged**